### PR TITLE
WIP: initial versioning work

### PR DESCRIPTION
**Pull request by mrengstrom**

### CODE REVIEW

File: app/utility/base_world.py [hunk 1]
```diff
@@ -109,6 +111,19 @@ async def walk_file_path(path, target):
                 return os.path.join(root, '%s.xored' % target)
         return None
 
+    @staticmethod
+    def get_version(path='.'):
+        ignore = ['/plugins/', '/.tox/']
```
Comment (privateducky) [inline app/utility/base_world.py:116]: I feel like this should return a tuple-- the version (number), and the hash, separately. we will display the former on the GUI whereas the latter is for validation only

Comment (privateducky) [reply app/utility/base_world.py:116]: ignore

File: static/js/shared.js [hunk 2]
```diff
@@ -137,9 +137,32 @@ window.onload = function checkBrowser(){
             if (scroll >= 100) sticky.addClass('.notice');
             else sticky.removeClass('.notice');
           });
+     }
+}
+
+function checkVersions(){
+    function evaluateVersions(data){
+        console.log(data)
```
Comment (privateducky) [inline static/js/shared.js:145]: remove log and comment

File: app/objects/c_plugin.py [hunk 3]
```diff
@@ -23,6 +23,7 @@ def __init__(self, name='virtual', description=None, address=None, enabled=False
         self.enabled = enabled
         self.data_dir = data_dir
         self.access = access if access else self.Access.APP
+        self.version = 'unknown'
```
Comment (privateducky) [inline app/objects/c_plugin.py:26]: I think this'd be better as None instead of 'unknown'

File: app/service/rest_svc.py [hunk 4]
```diff
@@ -237,6 +237,12 @@ async def validate(op):
             operation[0].autonomous = 0 if operation[0].autonomous else 1
             self.log.debug('Toggled operation=%s autonomous to %s' % (op_id, bool(autonomous)))
 
+    async def retrieve_versions(self):
+        app_svc = self.get_service('app_svc')
+        plugins = await self.get_service('data_svc').locate('plugins')
```
Comment (privateducky) [inline app/service/rest_svc.py:242]: I think you can kill this function

File: static/js/shared.js (hunk 2 above)
Comment (mrengstrom) [reply static/js/shared.js:145]: already removed locally, the function itself needs some work

File: app/service/rest_svc.py (hunk 4 above)
Comment (mrengstrom) [reply app/service/rest_svc.py:242]: removed on the next push

File: VERSION.txt [hunk 5]
```diff
@@ -0,0 +1 @@
+2.6.7-887a0e058b3ebeb1796535e4104e7726
```
Comment (privateducky) [inline VERSION.txt:1]: kill this file change

**Comment by mrengstrom** [comment]
rebased into a separate pr

**Pull request closed without merging**